src/StaticMappedResourceHandlerFactory.php

```diff
@@ -0,0 +1,164 @@
+ *             'gzip' => [
+ *                 'level' => 4, // Integer between 1 and 9 indicating compression level to use.
+ *                               // Values less than 1 disable compression.
+ *             ],
+ *             'directives' => [
+ *                 // Rules governing which server-side caching headers are emitted.
+ *                 // Each key must be a valid regular expression, and should match
+ *                 // typically only file extensions, but potentially full paths.
+ *                 // When a static resource matches, all associated rules will apply.
+ *                 'regex' => [
+ *                     'cache-control' => [
+ *                         // one or more valid Cache-Control directives:
+ *                         // - must-revalidate
+ *                         // - no-cache
+ *                         // - no-store
+ *                         // - no-transform
+ *                         // - public
+ *                         // - private
+ *                         // - max-age=\d+
+ *                     ],
+ *                     'last-modified' => bool, // Emit a Last-Modified header?
+ *                     'etag' => bool, // Emit an ETag header?
+ *                 ],
+ *             ],
+ *         ],
+ *     ],
+ * ],
+ * </code>
+ */
+class StaticMappedResourceHandlerFactory
+{
+    public function __invoke(ContainerInterface $container) : StaticMappedResourceHandler
+    {
+        $config = $container->get('config')['mezzio-swoole']['swoole-http-server']['static-files'] ?? [];
+        return new StaticMappedResourceHandler(
+            $container->get(StaticResourceHandler\FileLocationRepositoryInterface::class),
+            $this->configureMiddleware($config)
+        );
+    }
+
+    /**
+     * Prepare the list of middleware based on configuration provided.
+     *
+     * Examines the configuration provided and uses it to create the list of
+     * middleware to return. By default, the following are always present:
+     *
+     * - MethodNotAllowedMiddleware
+     * - OptionsMiddleware
+     * - HeadMiddleware
+     *
+     * If the clearstatcache-interval setting is present and non-false, it is
+     * used to seed a ClearStatCacheMiddleware instance.
+     *
+     * If any cache-control directives are discovered, they are used to seed a
+     * CacheControlMiddleware instance.
+     *
+     * If any last-modified directives are discovered, they are used to seed a
+     * LastModifiedMiddleware instance.
+     *
+     * If any etag directives are discovered, they are used to seed a
+     * ETagMiddleware instance.
+     *
+     * This method is marked protected to allow users to extend this factory
+     * in order to provide their own middleware and/or configuration schema.
+     *
+     * @return StaticResourceHandler\MiddlewareInterface[]
+     */
+    protected function configureMiddleware(array $config) : array
```

Comment:
Let's create either a trait that provides this method, or an abstract class that both factory implementations extend that provides it, so we can avoid duplication.